app/controllers/favorites_controller.rb

```diff
@@ -16,4 +16,15 @@ def update
     redirect_to "/pets/#{pet.id}"
   end
 
+  def destroy
+    if params[:pet_id] == nil
+      favorites.pets_favorite.clear
+    else
+      pet = Pet.find(params[:pet_id])
+      favorites.pets_favorite.delete(params[:pet_id])
+      flash[:notice] = "The pet has been removed from your Favorite Pets."
+    end
+    redirect_back(fallback_location: '/favorites')
+  end
+
```

Comment:
To know what is happening here, you'll also have to take a look at the routes. The routes I added are as follows;

- delete '/favorites/:pet_id', to: 'favorites#destroy'

When the site hits this route, it will skip down to the else statement and only remove the one pet since there is a pet_id present.

- delete '/favorites', to: 'favorites#destroy'

The site will only hit this route when the delete all button is clicked from the favorites index. This means that the first if statement is satisfied since params[:pet_id] == nil, and it will clear all pets from the favorites. 

The redirect_back is a fallback route that will only hit if there is not another route specified. I am not entirely positive as to how this works, but it was the only way to format this in order to get the flash message to appear whenever a pet is removed. Might be a good clarifying question for the instructor at our checkin tomorrow.

Reply:
The paths makes sense, but that redirect_back is confusing.  Is there another route that it could hit? Let's check in with the instructors about it.